src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -1623,8 +1638,32 @@ async def service_remote(middleware, service, verb, options):
 
 
 async def ready_system_sync_keys(middleware):
+
     if await middleware.call('failover.licensed'):
-        await middleware.call('failover.call_remote', 'failover.sync_keys_to_remote_node')
+
+        # when middlewared is setup, we only run this if we're the standby controller
+        if not await middleware.call('failover.is_backup_node'):
+            return
+
+        try:
+            await middleware.call('failover.call_remote', 'failover.sync_keys_to_remote_node')
+        except Exception as e:
+            if e.errno == CallError.ENOMETHOD:
+                # We're talking to an <= 11.3 system, so use the old API
```

Comment:
I am not sure if we need this in 2 places ? When backup node comes back online it should perform this sync. Do you think it's safe to remove the code in event.py or from here ?

Reply:
Spoke with William about that and to be safe, we need to keep it in both places. The event plugin is really the only way of knowing when we're the BACKUP controller.

We also need to sync keys on "setup" of middlewared so we'll leave it there too.

Reply:
In that case let's please have a common method for this which we can call in both places.

Reply:
A prime example of when the event plugin isn't used but the setup is used is if someone restarts middlewared on the backup controller.

Reply:
It might be nice to have a wrapper around `sync_keys_to_remote_node` which we could use in these two places and then remove down the road